rossum_api/elis_api_client_sync.py

```diff
@@ -75,29 +76,24 @@ def __init__(
         self.elis_api_client = ElisAPIClient(
             username, password, token, base_url, http_client, deserializer
         )
-
-        try:
-            self.event_loop = asyncio.get_running_loop()
-            if self.event_loop.is_running():
-                raise AsyncRuntimeError(
-                    "Event loop is present and already running, please use async version of the client"
-                )
-        except RuntimeError:
-            self.event_loop = asyncio.new_event_loop()
+        self.executor = ThreadPoolExecutor()
 
     def _iter_over_async(self, ait: AsyncIterator[T]) -> Iterator[T]:
-        ait = ait.__aiter__()
-        while True:
-            try:
-                obj = self.event_loop.run_until_complete(ait.__anext__())
-                yield obj
-            except StopAsyncIteration:
-                break
+        async def async_iter_to_list(ait: AsyncIterator[T]):
+            # TODO this materializes the whole generator into memory :(
+            return [obj async for obj in ait]
```

Comment:
This makes me sad but I didn't figure out how to use the previous pattern with ThreadPoolExectutor yet.

Reply:
I've managed to implement this via `queue.Queue` to communicate between asyncio in the thread and the sync code. Seems to work fine including error propagation (test for that added).